JsonPreprocessor/CJsonPreprocessor.py

```diff
@@ -2093,12 +2094,24 @@ def __handleLastElement(sInput : str) -> str:
                 sJsonDataUpdated = f"{sJsonDataUpdated}{line}\n"
         sJsonDataUpdated = regex.sub(r'\[\s+\'', '[\'', sJsonDataUpdated)
         sJsonDataUpdated = regex.sub(r'\'\s+\]', '\']', sJsonDataUpdated)
-        lKeyName = regex.findall(r'[,\s{]*("[^:,\n]*")\s*:\s*', sJsonDataUpdated)
+        lKeyName = regex.findall(r'[,\s{]*("[^"\n]*")\s*:\s*', sJsonDataUpdated)
+        tmpJsonDataUpdated = regex.sub(r":\s*\"[^\"]*\"", ": \"\"", sJsonDataUpdated)
+        tmpJsonDataUpdated = regex.sub(r"\[[^:]*:[^:]*\]", "[]", tmpJsonDataUpdated)
+        lKeyName = lKeyName + regex.findall(r'[,\s{]*(\${[^:,\n]+)\s*:\s*[^\]}]', tmpJsonDataUpdated)
         for key in lKeyName:
             if regex.match(r'^"\s+[^\s]+.+"$|^".+[^\s]+\s+"$', key):
                 newKey = '"' + key.strip('"').strip() + '"'
                 sJsonDataUpdated = sJsonDataUpdated.replace(key, newKey)
                 key = newKey
+            elif regex.match(r'^\s*\${.*$', key):
+                if key.count('${') != key.count('}'):
+                    errorMsg = f"Invalid syntax: '{key.strip()}' - The curly brackets are mismatch."
```

Comment:
Hi Son,

lots of improvements. Very good.

Tiny doubt about curious wording:
The curly brackets are mismatch.
The square brackets are mismatch.

Better:
The curly brackets do not match.

Reply:
Hello Hogler,

Thank you for the suggestion! I updated.